src/patterns/kinesis-lambda.ts

```diff
@@ -40,8 +39,8 @@ import type { StreamErrorHandlingProps, StreamProcessingProps } from "../utils/l
  * existingKinesisStream: { externalKinesisStreamName: "KinesisStreamFromAnotherStack" }
  * ```
  */
-export interface ExistingKinesisStream extends GuMigratingResource {
-  externalKinesisStreamName?: string;
+export interface ExistingKinesisStream {
```

Comment:
By dropping `GuMigratingResource` we're pushing the responsibility of persisting a logical ID for the kinesis stream to the user. In order to achieve this, we need to make the stream available (see the `public readonly` change below).

Reply:
This interface was previously dealing with 2 cases (but I think I did a bad job with the semantics!):

1. `existingKinesisStream` i.e. it belongs to _this_ CFN stack, but you had created it before you started using this pattern
2. `externalKinesisStream` i.e. it belongs to _another_ CFN stack 

Now that 1 has been removed, it feels a bit confusing to have the prop called `externalKinesisStreamName` (since it might not be external).

The good news here is that I don't think this pattern has many users yet (I think Journalism Dev were the first people to pick this up and that was only in the last fortnight or so), so I think we can break this API without causing too many problems.

Happy to chat about this one in person as I feel like I didn't get the API quite right in the first instance and it'd be nice to make this clearer at the second attempt!

Reply:
I think we can defer this in response to https://github.com/guardian/cdk/pull/1293, so will resolve this thread (i.e. no action needed here, IMO!).